### PR TITLE
fix: OP prompts for logout when no OP session

### PR DESCRIPTION
**Pull request by dopry**
The OAuth provider is prompting users who no longer have an user session with the OAuth Provider to logout of the OP. This happens in scenarios given the user has logged out of the OP directly or via another client. In cases where the user does not have a session on the OP we should not prompt them to log out of the OP as there is no session, but we should still clear out their tokens to terminate the session for the Application.

I also renamed a test that used expired_session in the name to without_op_session. I believe that was their original intent, @tonial can you confirm? 

## Checklist

- [X] PR only contains one change (considered splitting up PR)
- [X] unit-test added
- [X] documentation updated
- [x] `CHANGELOG.md` updated (only for user relevant changes)
- [X] author name in `AUTHORS`

### CODE REVIEW

File: tests/test_oidc_views.py [hunk 1]
```diff
@@ -577,29 +581,46 @@ def test_token_deletion_on_logout(oidc_tokens, logged_in_client, rp_settings):
...
+
     rsp = client.get(
         reverse("oauth2_provider:rp-initiated-logout"),
         data={
             "id_token_hint": oidc_tokens.id_token,
             "client_id": oidc_tokens.application.client_id,
         },
     )
-    assert rsp.status_code == 200
+    assert rsp.status_code == 302
```
Comment (n2ygk) [inline tests/test_oidc_views.py:599]: why is this changed from a 200 to a 302 redirect?

**Review by n2ygk — APPROVED** [review]

**Comment by tonial** [comment]
@dopry yes, that was the case i was covering
